google-cloud-nio/src/main/java/com/google/cloud/storage/contrib/nio/CloudStorageFileSystemProvider.java

```diff
@@ -267,8 +268,14 @@ && isNullOrEmpty(uri.getUserInfo()),
   @Override
   public CloudStoragePath getPath(URI uri) {
     initStorage();
-    return CloudStoragePath.getPath(
-        getFileSystem(CloudStorageUtil.stripPathFromUri(uri)), uri.getPath());
+    CloudStoragePath cloudStoragePath =
+        CloudStoragePath.getPath(
+            getFileSystem(CloudStorageUtil.stripPathFromUri(uri)), uri.getPath());
+    if (storage.get(cloudStoragePath.bucket()) == null) {
```

Comment:
It seems mildly silly to create the file system and then say we couldn't create the File System.   :)


I don't think users will be expecting this.  Normal file operations are allowed to describe Paths which do not exist. 

I.e
```
    Paths.get("/this/doesnt/exist");
```
is totally fine.  

I understand that there is some difference here because buckets are essentially their own file system, but I think users expect to treat it conceptually as as single gcs:// file system with many subfolders (buckets).

This also has the problem that buckets can be created or deleted while the program is running so there's no guarantee that it won't become invalid in the future.   

 It also precludes any ability to create a new bucket as part of this library, which is an obvious thing we might want to be able to do in the future.

Reply:
Does this also add an additional requirement on the (possibly deprecated) permission "storage.buckets.get"?